com.hazelcast.jet.pipeline.OrderedProcessingMultipleMemberTest.multiple_nodes [HZ-1827]

_master_ (commit 2be3ee8d641129536e7c12fef0d48100cc2e092a)

Failed on CorrettoJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-CorrettoJDK17/39/testReport/com.hazelcast.jet.pipeline/OrderedProcessingMultipleMemberTest/multiple_nodes_2__transform_map_using_imap_high_/

Failed in variant `[2: transform=map-using-imap-high]`.

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 
Job was expected to complete with AssertionCompletedException, but completed with: com.hazelcast.jet.JetException: java.lang.IllegalStateException: Coordinator [127.0.0.1]:5701 cannot initialize job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001. Reason: it is not the master, the master is [127.0.0.1]:5702
	at com.hazelcast.jet.impl.JobExecutionService.assertIsMaster(JobExecutionService.java:403)
	at com.hazelcast.jet.impl.JobExecutionService.addExecutionContext(JobExecutionService.java:376)
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:323)
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.doRun(InitExecutionOperation.java:97)
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:54)
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:471)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

	at com.hazelcast.jet.core.test.JetAssert.assertTrue(JetAssert.java:38)
	at com.hazelcast.jet.pipeline.OrderedProcessingMultipleMemberTest.multiple_nodes(OrderedProcessingMultipleMemberTest.java:280)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details>

<details><summary>Standard output:</summary>

```
23:15:31,659 DEBUG || - [JobRepository] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
23:15:31,669 DEBUG || - [JobClassLoaderService] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Creating job classLoader for job 081d-ebd4-20c4-0001
23:15:31,669 DEBUG || - [JobClassLoaderService] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Create processor classloader map for job 081d-ebd4-20c4-0001
23:15:31,669 DEBUG || - [Planner] hz.pedantic_chandrasekhar.cached.thread-4 - Watermarks in the pipeline will be throttled to 1
23:15:31,671  INFO || - [JobCoordinationService] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Starting job 081d-ebd4-20c4-0001 based on submit request
23:15:31,672  INFO || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Didn't find any snapshot to restore for job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001
23:15:31,672  INFO || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start executing job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001, execution graph in DOT format:
digraph DAG {
	"mapJournalSource(test-map-2)" [localParallelism=2];
	"mapUsingIMap" [localParallelism=2];
	"map-stateful-keyed" [localParallelism=2];
	"assertCollectedEventually" [localParallelism=1];
	"mapJournalSource(test-map-2)" -> "mapUsingIMap" [label="isolated", queueSize=1024];
	"mapUsingIMap" -> "map-stateful-keyed" [label="distributed-partitioned", queueSize=1024];
	"map-stateful-keyed" -> "assertCollectedEventually" [label="distributed-partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com/ to visualize the printed graph.
23:15:31,672 DEBUG || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Building execution plan for job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001
23:15:31,672 DEBUG || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Built execution plans for job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001
23:15:31,672 DEBUG || - [InitExecutionOperation] hz.eager_chandrasekhar.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initializing execution plan for job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001 from [127.0.0.1]:5701
23:15:31,672 DEBUG || - [InitExecutionOperation] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing execution plan for job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001 from [127.0.0.1]:5701
23:15:31,672 DEBUG || - [JobClassLoaderService] hz.eager_chandrasekhar.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Creating job classLoader for job 081d-ebd4-20c4-0001
23:15:31,672 DEBUG || - [JobClassLoaderService] hz.eager_chandrasekhar.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Create processor classloader map for job 081d-ebd4-20c4-0001
23:15:31,673  WARN || - [NodeMulticastListener] hz.eager_chandrasekhar.MulticastThread - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] New join request has been received from current master address. The UUID in the join request (cadb25fe-b130-4604-8b7c-95414c1c11ab) is different from the known master one (082dc3f5-b171-47c4-9add-a88aa5cdedb9). Suspecting the master address: [127.0.0.1]:5701
23:15:31,673  WARN || - [MembershipManager] hz.eager_chandrasekhar.MulticastThread - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Member [127.0.0.1]:5701 - 082dc3f5-b171-47c4-9add-a88aa5cdedb9 is suspected to be dead for reason: New join request has been received from current master address. The UUID in the join request (cadb25fe-b130-4604-8b7c-95414c1c11ab) is different from the known master one (082dc3f5-b171-47c4-9add-a88aa5cdedb9). Suspecting the master address: [127.0.0.1]:5701
23:15:31,673  INFO || - [MembershipManager] hz.eager_chandrasekhar.MulticastThread - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Starting mastership claim process...
23:15:31,673  INFO || - [MembershipManager] hz.eager_chandrasekhar.MulticastThread - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Local MembersView{version=2, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=082dc3f5-b171-47c4-9add-a88aa5cdedb9, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5702, uuid=7727f4fb-bd96-4df2-9c7b-46be4f3d45f9, liteMember=false, memberListJoinVersion=2}]} with suspected members: [Member [127.0.0.1]:5701 - 082dc3f5-b171-47c4-9add-a88aa5cdedb9] and initial addresses to ask: []
23:15:31,673 DEBUG || - [JobClassLoaderService] hz.eager_chandrasekhar.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=081d-ebd4-20c4-0001
23:15:31,673  INFO || - [JobExecutionService] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution plan for jobId=081d-ebd4-20c4-0001, jobName='081d-ebd4-20c4-0001', executionId=081d-ebd4-20c5-0001 initialized
23:15:31,673 ERROR || - [InitExecutionOperation] hz.eager_chandrasekhar.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Coordinator [127.0.0.1]:5701 cannot initialize job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001. Reason: it is not the master, the master is [127.0.0.1]:5702
java.lang.IllegalStateException: Coordinator [127.0.0.1]:5701 cannot initialize job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001. Reason: it is not the master, the master is [127.0.0.1]:5702
	at com.hazelcast.jet.impl.JobExecutionService.assertIsMaster(JobExecutionService.java:403) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.addExecutionContext(JobExecutionService.java:376) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:323) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.doRun(InitExecutionOperation.java:97) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:54) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:471) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
23:15:31,673  INFO || - [TcpServerConnection] hz.eager_chandrasekhar.cached.thread-5 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=1, /0.0.0.0:46850->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=082dc3f5-b171-47c4-9add-a88aa5cdedb9, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
23:15:31,673  INFO || - [TcpServerConnection] hz.pedantic_chandrasekhar.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:46850, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=7727f4fb-bd96-4df2-9c7b-46be4f3d45f9, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
23:15:31,673  INFO || - [PartitionStateManager] hz.eager_chandrasekhar.cached.thread-5 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 082dc3f5-b171-47c4-9add-a88aa5cdedb9
23:15:31,673 DEBUG || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Init of job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001 has failures: [[127.0.0.1]:5702=java.lang.IllegalStateException: Coordinator [127.0.0.1]:5701 cannot initialize job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001. Reason: it is not the master, the master is [127.0.0.1]:5702]
23:15:31,673  INFO || - [ClusterService] hz.eager_chandrasekhar.cached.thread-5 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5702 - 7727f4fb-bd96-4df2-9c7b-46be4f3d45f9 this
]

23:15:31,673  INFO || - [TransactionManagerService] hz.eager_chandrasekhar.cached.thread-3 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: 082dc3f5-b171-47c4-9add-a88aa5cdedb9
23:15:31,673  INFO || - [MembershipManager] hz.eager_chandrasekhar.cached.thread-5 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Mastership is claimed with: MembersView{version=3, members=[MemberInfo{address=[127.0.0.1]:5702, uuid=7727f4fb-bd96-4df2-9c7b-46be4f3d45f9, liteMember=false, memberListJoinVersion=2}]}
23:15:31,674  INFO || - [TcpServerConnector] hz.pedantic_chandrasekhar.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
23:15:31,674 DEBUG || - [JobExecutionService] hz.pedantic_chandrasekhar.async.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001 calling completeExecution because execution terminated before it started
23:15:31,674 DEBUG || - [JobExecutionService] hz.pedantic_chandrasekhar.async.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Completed execution of job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001
23:15:31,674  INFO || - [TcpServerConnection] hz.eager_chandrasekhar.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:42917
23:15:31,674  INFO || - [TcpServerConnection] hz.pedantic_chandrasekhar.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:42917 and /127.0.0.1:5702
23:15:31,924  INFO || - [InternalPartitionService] hz.eager_chandrasekhar.migration - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: -5749966051988468662
23:15:31,924  INFO || - [InternalPartitionService] hz.eager_chandrasekhar.migration - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Most recent partition table is determined.
23:15:31,924  INFO || - [InternalPartitionService] hz.eager_chandrasekhar.migration - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Applying the most recent of partition state...
23:15:31,924  WARN || - [InternalPartitionService] hz.eager_chandrasekhar.migration - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Following unknown addresses are found in partition table sent from master[[127.0.0.1]:5702]. (Probably they have recently joined or left the cluster.) {
	[127.0.0.1]:5701 - 082dc3f5-b171-47c4-9add-a88aa5cdedb9
}
23:15:31,925  INFO || - [MigrationManager] hz.eager_chandrasekhar.migration - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Partition balance is ok, no need to repartition.
23:15:33,282  INFO || - [ExplicitSuspicionOp] hz.pedantic_chandrasekhar.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5702
23:15:33,282  INFO || - [TcpServerConnection] hz.pedantic_chandrasekhar.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=2, /0.0.0.0:42917->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=7727f4fb-bd96-4df2-9c7b-46be4f3d45f9, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
23:15:33,282  INFO || - [TcpServerConnection] hz.eager_chandrasekhar.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:42917, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=082dc3f5-b171-47c4-9add-a88aa5cdedb9, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
23:15:33,282  INFO || - [MembershipManager] hz.pedantic_chandrasekhar.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 7727f4fb-bd96-4df2-9c7b-46be4f3d45f9
23:15:33,282  INFO || - [PartitionStateManager] hz.pedantic_chandrasekhar.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 7727f4fb-bd96-4df2-9c7b-46be4f3d45f9
23:15:33,282  INFO || - [ClusterService] hz.pedantic_chandrasekhar.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - 082dc3f5-b171-47c4-9add-a88aa5cdedb9 this
]

23:15:33,282  INFO || - [TransactionManagerService] hz.pedantic_chandrasekhar.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 7727f4fb-bd96-4df2-9c7b-46be4f3d45f9
23:15:33,282 ERROR || - [MasterJobContext] ForkJoinPool.commonPool-worker-261 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001: some TerminateExecutionOperation invocations failed, execution might remain stuck: [MemberInfo{address=[127.0.0.1]:5701, uuid=082dc3f5-b171-47c4-9add-a88aa5cdedb9, liteMember=false, memberListJoinVersion=1}=null, MemberInfo{address=[127.0.0.1]:5702, uuid=7727f4fb-bd96-4df2-9c7b-46be4f3d45f9, liteMember=false, memberListJoinVersion=2}=com.hazelcast.core.MemberLeftException: Member [127.0.0.1]:5702 - 7727f4fb-bd96-4df2-9c7b-46be4f3d45f9 has left cluster!]
23:15:33,282 DEBUG || - [JobClassLoaderService] hz.pedantic_chandrasekhar.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=081d-ebd4-20c4-0001
23:15:33,283 ERROR || - [MasterJobContext] hz.pedantic_chandrasekhar.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001 failed
	Start time: 2022-06-02T23:15:31.671
	Duration: 00:00:01.611
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
java.lang.IllegalStateException: Coordinator [127.0.0.1]:5701 cannot initialize job 081d-ebd4-20c4-0001, execution 081d-ebd4-20c5-0001. Reason: it is not the master, the master is [127.0.0.1]:5702
	at com.hazelcast.jet.impl.JobExecutionService.assertIsMaster(JobExecutionService.java:403) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.addExecutionContext(JobExecutionService.java:376) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:323) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.doRun(InitExecutionOperation.java:97) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:54) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:471) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
23:15:33,283 DEBUG || - [JobCoordinationService] hz.pedantic_chandrasekhar.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '081d-ebd4-20c4-0001', execution 081d-ebd4-20c5-0001 is completed
23:15:33,283  INFO || - [MigrationManager] hz.pedantic_chandrasekhar.migration - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Partition balance is ok, no need to repartition.
BuildInfo right after multiple_nodes[2: transform=map-using-imap-high](com.hazelcast.jet.pipeline.OrderedProcessingMultipleMemberTest): BuildInfo{version='5.2-SNAPSHOT', build='20220602', buildNumber=20220602, revision=2be3ee8, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'multiple_nodes[2: transform=map-using-imap-high](com.hazelcast.jet.pipeline.OrderedProcessingMultipleMemberTest):'
23:15:30, accumulated pauses: 1484 ms, max pause: 1460 ms, pauses over 1000 ms: 1
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/jet/pipeline/OrderedProcessingMultipleMemberTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/pipeline/OrderedProcessingMultipleMemberTest.java
@@ -40,7 +40,6 @@ import org.junit.runners.Parameterized.Parameter;
 import org.junit.runners.Parameterized.Parameters;
 import org.junit.runners.Parameterized.UseParametersRunnerFactory;
 
-import java.io.Serializable;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Map;
@@ -64,7 +63,7 @@ import static com.hazelcast.jet.core.test.JetAssert.fail;
 @RunWith(HazelcastParametrizedRunner.class)
 @UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
 @Category(QuickTest.class)
-public class OrderedProcessingMultipleMemberTest extends SimpleTestInClusterSupport implements Serializable {
+public class OrderedProcessingMultipleMemberTest extends SimpleTestInClusterSupport {
 
     // Used to set the LP of the stage with the higher value than upstream parallelism
     private static final int HIGH_LOCAL_PARALLELISM = 11;
@@ -244,7 +243,7 @@ public class OrderedProcessingMultipleMemberTest extends SimpleTestInClusterSupp
         StreamStage<Map.Entry<Long, Long>> applied = srcStage.apply(transform);
 
         applied.groupingKey(Map.Entry::getKey)
-                .mapStateful(() -> create(keyCount), this::orderValidator)
+                .mapStateful(() -> create(keyCount), OrderedProcessingMultipleMemberTest::orderValidator)
                 .writeTo(AssertionSinks.assertCollectedEventually(60,
                         list -> {
                             assertTrue("when", itemCount <= list.size());
@@ -277,7 +276,7 @@ public class OrderedProcessingMultipleMemberTest extends SimpleTestInClusterSupp
         return state;
     }
 
-    private boolean orderValidator(LongAccumulator[] s, Long key, Map.Entry<Long, Long> entry) {
+    private static boolean orderValidator(LongAccumulator[] s, Long key, Map.Entry<Long, Long> entry) {
         LongAccumulator acc = s[key.intValue()];
         long value = entry.getValue();
         if (acc.get() >= value) {

--- a/hazelcast/src/test/java/com/hazelcast/jet/pipeline/OrderedProcessingMultipleMemberTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/pipeline/OrderedProcessingMultipleMemberTest.java
@@ -18,14 +18,12 @@ package com.hazelcast.jet.pipeline;
 
 import com.hazelcast.config.Config;
 import com.hazelcast.config.EventJournalConfig;
-import com.hazelcast.core.Hazelcast;
 import com.hazelcast.function.FunctionEx;
 import com.hazelcast.function.PredicateEx;
-import com.hazelcast.jet.JetInstance;
 import com.hazelcast.jet.Job;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
 import com.hazelcast.jet.Traversers;
 import com.hazelcast.jet.accumulator.LongAccumulator;
-import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.processor.Processors;
 import com.hazelcast.jet.pipeline.test.AssertionCompletedException;
 import com.hazelcast.jet.pipeline.test.AssertionSinks;
@@ -33,7 +31,6 @@ import com.hazelcast.map.IMap;
 import com.hazelcast.test.HazelcastParametrizedRunner;
 import com.hazelcast.test.HazelcastSerialParametersRunnerFactory;
 import com.hazelcast.test.annotation.QuickTest;
-import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -67,7 +64,7 @@ import static com.hazelcast.jet.core.test.JetAssert.fail;
 @RunWith(HazelcastParametrizedRunner.class)
 @UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
 @Category(QuickTest.class)
-public class OrderedProcessingMultipleMemberTest extends JetTestSupport implements Serializable {
+public class OrderedProcessingMultipleMemberTest extends SimpleTestInClusterSupport implements Serializable {
 
     // Used to set the LP of the stage with the higher value than upstream parallelism
     private static final int HIGH_LOCAL_PARALLELISM = 11;
@@ -77,7 +74,6 @@ public class OrderedProcessingMultipleMemberTest extends JetTestSupport implemen
     private static final String JOURNALED_MAP_PREFIX = "test-map-";
 
     private static Pipeline p;
-    private static JetInstance[] instances;
 
     @Parameter(value = 0)
     public int idx;
@@ -96,22 +92,13 @@ public class OrderedProcessingMultipleMemberTest extends JetTestSupport implemen
                 .getEventJournalConfig();
         eventJournalConfig.setEnabled(true);
         eventJournalConfig.setCapacity(30000); // 30000/271 ~= 111 item per partition
-        instances = new JetInstance[INSTANCE_COUNT];
-        for (int i = 0; i < INSTANCE_COUNT; i++) {
-            instances[i] = (JetInstance) Hazelcast.newHazelcastInstance(config).getJet();
-        }
+
+        initialize(INSTANCE_COUNT, config);
     }
 
     @Before
     public void setup() {
         p = Pipeline.create().setPreserveOrder(true);
-    }
-
-    @AfterClass
-    public static void cleanup() {
-        for (int i = 0; i < INSTANCE_COUNT; i++) {
-            instances[i].shutdown();
-        }
     }
 
     @Parameters(name = "{index}: transform={2}")
@@ -265,12 +252,12 @@ public class OrderedProcessingMultipleMemberTest extends JetTestSupport implemen
                         }
                 ));
 
-        IMap<Long, Long> testMap = instances[0].getMap(mapName);
+        IMap<Long, Long> testMap = instance().getMap(mapName);
         LongStream.range(0, itemCount)
                 .boxed()
                 .forEachOrdered(i -> testMap.put(i % keyCount, i));
 
-        Job job = instances[0].newJob(p);
+        Job job = instance().getJet().newJob(p);
         try {
             job.join();
             fail("Job should have completed with an AssertionCompletedException, but completed normally");

